go/test/action.yml

```diff
@@ -1,17 +1,26 @@
 name: 'go-test'
 description: 'Custom GitHub action that runs all Golang tests within a repository and saves the test + coverage reports as artifacts'
 
+inputs:
+  TAGS:
+    description: Set of tags to run tests with.
+    required: false
+  TIMEOUT:
+    description: A timeout value. If tests take longer than this, they will fail. Set to an empty string to disable.
+    required: false
+    default: "30"
+
 runs:
   using: 'composite'
   steps:
     - name: Running go tests
       shell: bash
-      run: go test ./... -coverprofile=coverage.out -json -timeout 30s > test-report.out
+      run: ${GITHUB_ACTION_PATH}/test.sh "${{ inputs.TAGS }}" "${{ inputs.TIMEOUT }}"
     - name: Uploading test report
       uses: actions/upload-artifact@v3
       with:
-        name: "${{ github.sha }}-test-report.out"
-        path: "./test-report.out"
+        name: "${{ github.sha }}-report.xml"
+        path: "./report.xml"
```

Comment:
or we keep the name as `... .out` if that works

Reply:
Honestly I think sonarqube needs some inputs as well rather than assuming it knows what the files are called. This way we can reuse it for Ruby/other languages.